packages/medusa/src/api-v2/auth/[scope]/[authProvider]/callback/route.ts

```diff
@@ -0,0 +1,42 @@
+import { AuthenticationInput, IAuthModuleService } from "@medusajs/types"
+import { MedusaRequest, MedusaResponse } from "../../../../../types/routing"
+
+import { MedusaError } from "@medusajs/utils"
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+
+export const GET = async (req: MedusaRequest, res: MedusaResponse) => {
+  const { scope, authProvider } = req.params
+
+  const service: IAuthModuleService = req.scope.resolve(
+    ModuleRegistrationName.AUTH
+  )
+
+  const authData = {
+    url: req.url,
+    headers: req.headers,
+    query: req.query,
+    body: req.body,
+    authScope: scope,
+  } as AuthenticationInput
+
+  const authResult = await service.validateCallback(authProvider, authData)
+
+  const { success, error, authUser, location } = authResult
+  if (location) {
```

Comment:
if there is a location (the redirect could be configured) then we don't check any success or failure?

Reply:
yes, we could validate to ensure that `success` is required if you think that's appropriate? 😄 just figured that in case of a redirect you would want that no matter what

Reply:
in general you have a success redirect and a failure redirect that eventually could have custom query parameters, let see that later